### DISCUSSION TITLE
Exe opening then closing!

### DISCUSSION THREAD

**Discussion by AoAOrnstein** [Pryaxis/TShock#2846] [Q&A]
I'm not entirely sure on what Im doing wrong as Ive got everything required to run the program, followed the install instructions as I've hosted TShock before in the past but on a different PC but when I go to launch the exe it opens then instantly closes. Launch the normal TerrariaServer.exe and opens it normally without TShock loading, tried digging around through new and old posts trying to find my problem but I have no idea on what could be causing it!

**Comment by sgkoishi** [comment]
Probably you need [this](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.11-windows-x64-installer).

**Reply by AoAOrnstein** [reply]
I've got it up to date, thanks for the fast reply

Did a restart after uninstalling and reinstalling to try that and get the same result, wondering if maybe I have a program conflicting or just bad juju on my end

**Reply by AoAOrnstein** [reply]
Still not sure on what is causing it, toyed around with it a bit more and can't figure out what's causing it

**Reply by sgkoishi** [reply]
Can you run `dotnet --list-runtimes`? It should contains a 6.0 
Or, try the [latest dev version](https://github.com/Pryaxis/TShock/actions/workflows/ci-otapi3.yml?query=branch%3Ageneral-devel): currently it contains an installer/launcher and should download necessary things automatically.

**Reply by AoAOrnstein** [reply]
Ok, ran the list and it does show AspNetCore 6.0.11 and the regular NetCore 3.1.8
I'll try the dev version here in a moment

**Reply by AoAOrnstein** [reply]
Sorry for the long period in replies, internet has been on the fritz the past few days but I sat down and tried it last night and if I launch the installer from the beta branch it loads TShock and everything normally but if I use the server.exe it opens then closes, making progress!
